app/models/grant_submission/response.rb

```diff
@@ -185,5 +186,9 @@ def parse_raw_value_as_a_number(raw_value)
         end
       end
     end
+
+    def attachment_is_valid
+      errors.add(:document, :not_a_file_upload, question: question) unless question.response_type == 'file_upload'
```

Comment:
Maybe use an Enum or constant for response_type.  Having hardcoded strings in more than one place is not ideal.